### PR TITLE
check quota available before presenting options

### PR DESCRIPTION
**Pull request by laurafitzgerald**
https://issues.redhat.com/browse/SDA-6854

Changes include a function to get the quota_cost by addonname aka `addon|addon-<addononname>`
Check the orgs quota is not 0 before allowing install to progress
Check the quota when param is of type resource, only offer options that there is quota for 
on edit this includes the value of the installation being edited.

### CODE REVIEW

File: cmd/edit/addon/cmd.go [hunk 1]
```diff
@@ -75,6 +76,12 @@ func run(cmd *cobra.Command, argv []string) {
 		os.Exit(1)
 	}
 
+	quotaCost, err := r.OCMClient.GetAddonParameterQuotaCost(addOnID)
+	if err != nil {
+		r.Reporter.Warnf("%s", err)
+		os.Exit(0)
```
Comment (vkareh) [inline cmd/edit/addon/cmd.go:82]: If you're exiting due to an error it should be
```
		r.Reporter.Errorf("%s", err)
		os.Exit(1)
```

File: cmd/install/addon/cmd.go [hunk 2]
```diff
@@ -117,6 +118,17 @@ func run(cmd *cobra.Command, argv []string) {
 		os.Exit(1)
 	}
 
+	quotaCost, err := r.OCMClient.GetAddonParameterQuotaCost(addOnID)
+	if err != nil {
+		r.Reporter.Warnf("%s", err)
+		os.Exit(0)
```
Comment (vkareh) [inline cmd/install/addon/cmd.go:124]: Same here

File: cmd/edit/addon/cmd.go [hunk 3]
```diff
@@ -125,16 +131,35 @@ func run(cmd *cobra.Command, argv []string) {
...
+			availableQuota := quotaCost.Allowed() - quotaCost.Consumed() + paramValueInt
+			valueInt, err := strconv.Atoi(opt.Value())
+			if err != nil {
+				r.Reporter.Errorf("%s", err)
+				os.Exit(1)
+			}
+			if valueInt <= availableQuota {
+				options = append(options, opt.Name())
+				values = append(values, opt.Value())
+			}
```
Comment (ciaranRoche) [inline cmd/edit/addon/cmd.go:162]: The more I am thinking about this, we are assuming the quota does not change between when we fetch the quota and the params, versus when the user actually sends the request. 

I think we should maintain the current logic, give the user all options and allow them to make the request, and let AMS make the decision to reject or accept it. 

WDYT

**Review by ciaranRoche — COMMENTED** [review]
Lets set up a call and have a chat about this ask from QE?

File: cmd/install/addon/cmd.go [hunk 4]
```diff
@@ -117,6 +118,17 @@ func run(cmd *cobra.Command, argv []string) {
...
+	}
+	availableQuota := quotaCost.Allowed() - quotaCost.Consumed()
+	if availableQuota < 1 {
+		r.Reporter.Warnf("Your organisation account does not have any quota to install the add on %s", addOnID)
+		os.Exit(0)
+	}
```
[the remark this answers is not in the document]
Comment (vkareh) [reply cmd/install/addon/cmd.go:130]: At the very least, this decision should be done inside `pkg/ocm` in a "HasAddOnQuota" function, not in the `cmd/`, but I agree, maybe we need a "has quota" endpoint in AMS.

@ciaranRoche do you think we can we use the addon inquiries endpoint for this?

**Pull request closed without merging**